aioxmpp/im/archive.py

```diff
@@ -0,0 +1,78 @@
+########################################################################
+# File name: archive.py
+# This file is part of: aioxmpp
+#
+# LICENSE
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+########################################################################
+import asyncio
+import collections
+
+
+class Pagination:
```

Comment:
Hah, I couldn’t even remember that this class was in the PR. Actually I may move this to another branch; those are intended to be used with MAM, so they don’t serve a purpose yet.

That interface definitely will need some work in any case.